include/simdjson/builtin.h

```diff
@@ -29,6 +29,10 @@ namespace simdjson {
    * code that uses it) will use westmere.
    */
   namespace builtin = SIMDJSON_BUILTIN_IMPLEMENTATION;
+  /**
```

Comment:
Hmm, I wonder if we should have `haswell::implementation_name`, `westmere::implementation_name`, etc. instead--then the user can say simdjson::builtin::implementation_name and they can also do it generically.

Reply:
Ok. Let me try.

Reply:
Perhaps better, what about `simdjson::builtin_implementation`? Then you can do `simdjson::builtin_implementation->name()`, as well as call other methods on the implementation.

Reply:
(Since you can already do implementation->name(), simdjson::builtin_implementation->name() seems like the best of all worlds. Or possibly simdjson::implementations::builtin->name(), I suppose.

Reply:
Ok. Done and documented.